lib/src/cpp/utils/ParseXMLString.cpp

```diff
@@ -57,7 +57,7 @@ void ParseXMLString::writeChars(
         memcpy(&pTmp[m_nTotal], toWrite, count);
 
         m_nTotal += count;
-        m_pBuffer = pTmp;
+        m_pBuffer = reinterpret_cast<unsigned char*>(pTmp);
```

Comment:
Try avoiding the `reinterpret_cast`. Define instead `pTmp` as an unsigned char vector.

```
unsigned char* pTmp = new unsigned char[m_nTotal + count + 1];
```

As per the documentation `It is purely a compile-time directive which instructs the compiler to treat expression as if it had the type new-type.` In this case, it does not seem to be dangerous because both types have the same size and 2's complement arithmetic is probably used. However, I think it is best to avoid this potential dangerous issues if we can redefined the temporal buffer to be the same type as the buffer.